pkg/scanners/scannerv4/vm_convert.go

```diff
@@ -0,0 +1,166 @@
+			result = append(result, storage.VirtualMachineScan_UNSET)
+		case v4.VulnerabilityReport_NOTE_OS_UNKNOWN:
+			result = append(result, storage.VirtualMachineScan_OS_UNKNOWN)
+		case v4.VulnerabilityReport_NOTE_OS_UNSUPPORTED:
+			result = append(result, storage.VirtualMachineScan_OS_UNSUPPORTED)
+		default:
+			result = append(result, storage.VirtualMachineScan_UNSET)
+		}
+	}
+	return result
+}
+
+func toVirtualMachineComponents(r *v4.VulnerabilityReport) []*storage.EmbeddedVirtualMachineScanComponent {
+	packages := r.GetContents().GetPackages()
+	result := make([]*storage.EmbeddedVirtualMachineScanComponent, 0, len(packages))
+	for _, pkg := range packages {
+		vulnerabilityIDs := r.GetPackageVulnerabilities()[pkg.GetId()].GetValues()
+		vulnerabilitiesByID := r.GetVulnerabilities()
+		component := &storage.EmbeddedVirtualMachineScanComponent{
+			Name:    pkg.GetName(),
+			Version: pkg.GetVersion(),
+			// Architecture ?
+			Vulnerabilities: toVirtualMachineScanComponentVulnerabilities(vulnerabilitiesByID, vulnerabilityIDs),
+		}
+		result = append(result, component)
+	}
+	return result
+}
+
+func toVirtualMachineScanComponentVulnerabilities(
+	vulnerabilitiesByID map[string]*v4.VulnerabilityReport_Vulnerability,
+	vulnerabilityIDs []string,
+) []*storage.VirtualMachineVulnerability {
+	result := make([]*storage.VirtualMachineVulnerability, 0, len(vulnerabilityIDs))
+	processedVulnerabilityIDs := set.NewStringSet()
+	for _, vulnerabilityID := range vulnerabilityIDs {
+		if shouldProcess := processedVulnerabilityIDs.Add(vulnerabilityID); !shouldProcess {
+			continue
+		}
+		reportVulnerability, found := vulnerabilitiesByID[vulnerabilityID]
+		if !found {
+			continue
+		}
+		vulnerability := &storage.VirtualMachineVulnerability{
+			CveBaseInfo: &storage.VirtualMachineCVEInfo{
+				Cve:         reportVulnerability.GetName(),
+				Summary:     reportVulnerability.GetDescription(),
+				Link:        link(reportVulnerability.GetLink()),
+				PublishedOn: reportVulnerability.GetIssued(),
+				Advisory:    toVirtualMachineAdvisory(reportVulnerability.GetAdvisory()),
+			},
+			Severity: normalizedSeverity(reportVulnerability.GetNormalizedSeverity()),
+		}
+		vulnerabilityMetrics := reportVulnerability.GetCvssMetrics()
+		if err := setVirtualMachineScoresAndScoreVersions(vulnerability, vulnerabilityMetrics); err != nil {
+			utils.Should(err)
+		}
+
+		if reportVulnerability.GetFixedInVersion() != "" {
+			vulnerability.SetFixedBy = &storage.VirtualMachineVulnerability_FixedBy{
+				FixedBy: reportVulnerability.GetFixedInVersion(),
+			}
+		}
+
+		result = append(result, vulnerability)
+	}
+	return result
+}
+
+func toVirtualMachineAdvisory(
+	advisory *v4.VulnerabilityReport_Advisory,
+) *storage.VirtualMachineAdvisory {
+	if advisory == nil {
+		return nil
+	}
+	return &storage.VirtualMachineAdvisory{
+		Name: advisory.GetName(),
+		Link: advisory.GetLink(),
+	}
+}
+
+func setVirtualMachineScoresAndScoreVersions(
```

Comment:
I never dug into the scores and the calculations, so it is hard to review for me.